### ISSUE TITLE
How to get value of field in 'part' event

### ISSUE THREAD

**Issue by yanyu0517** [pillarjs/multiparty#186]
Hi,

I upload a picture and with a field named 'type' to server. The 'type' field has different value. Like the below:
![image](https://user-images.githubusercontent.com/5326851/34195408-5e01cc6c-e598-11e7-9fe3-988e8cc7f66c.png)
I used 'part' event to recevie stream in server, and get name filed ('type').
But how to get value of 'type' field in 'part' event?
My code in server
![image](https://user-images.githubusercontent.com/5326851/34195500-d77e6f5a-e598-11e7-89f9-fbdd0b755851.png)

**Comment by dougwilson** [comment]
Hi @yanyu0517 you get the value by reading the `part` that does not have a filename. In your code example above, you are just ignoring it. Just read the data instead of calling `formPart.result()` to get the value 👍

**Comment by yanyu0517** [comment]
I really not very clear how to get the value. 

```
if (!formPart.filename) {
    console.log('got field named ', formPart.name) // output 'type'
    // How can I get the value of 'type' field?
    // How to read the data instead of call formPart.resume()?
    formPart.resume();
}

```
Can you write an example for me?

Thank you

**Comment by dougwilson** [comment]
Hi @yanyu0517 the `formPart` variable in your example is a `ReadableStream` in Node.js, so you would get the contents just like you would for any Readable Stream. See the Node.js documentation: https://nodejs.org/dist/latest-v8.x/docs/api/stream.html#stream_readable_streams

If you're not familiar with how to do this, I suggest not trying to, because it's not simple and have various edge cases. You can instead listen for the `field` event (https://github.com/pillarjs/multiparty#field-name-value) in which you'll get the field content as a string already read for you.

**Comment by yanyu0517** [comment]
Thank you for your replay. I am not clear with the event execution order, is 'part' event always after the 'field' event executed? I find a closed issue #159 , but not any result.

**Comment by dougwilson** [comment]
Hi @yanyu0517 the events are fired in the order they are in your payload. Assuming your payload is the same as your screenshot at the top, you'll first get a field event and then a part event and then a close event.
